be/src/olap/compaction.cpp

```diff
@@ -186,7 +186,7 @@ Status Compaction::merge_input_rowsets() {
     Status res;
     {
         SCOPED_TIMER(_merge_rowsets_latency_timer);
-        if (_is_vertical) {
+        if (_is_vertical || !_tablet->tablet_schema()->cluster_key_idxes().empty()) {
```

Comment:
if non-vertical is configed and table has cluster key, just skip compaction by return error.